packages/squiggle-lang/src/PointSet/PointSetDist_Scoring.ts

```diff
@@ -137,19 +91,19 @@ export const WithScalarAnswer = {
       const density = estimatePdf(answer);
       if (density === undefined) {
         return Result.Err(PdfInvalidError);
+      } else if (density < 0) {
+        return Result.Err(PdfInvalidError);
+      } else if (density === 0) {
+        return Result.Ok(Infinity);
       } else {
-        if (density < 0) {
-          return Result.Err(PdfInvalidError);
-        } else if (density === 0) {
-          return Result.Ok(Infinity);
-        } else {
-          return Result.Ok(-logFn(density));
-        }
+        return Result.Ok(-logFn(density));
       }
     };
     const estimatePdf = (x: number) => {
-      if (isContinuous(estimate) || isDiscrete(estimate)) {
-        return WithScalarAnswer.sum(estimate.xToY(x));
+      if (estimate.toContinuous().isEmpty()) {
+        return WithScalarAnswer.sum(estimate.toDiscrete().xToY(x));
+      } else if (estimate.toDiscrete().isEmpty()) {
+        return WithScalarAnswer.sum(estimate.toContinuous().xToY(x));
```

Comment:
This code is awkward, but it works.

Ideally, I would refactor this entire file in a different style (half of it is legacy from Rescript), but I ran out of time.

Reply:
Makes sense. I prefer doing this in small chunks, like you did.